pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -392,10 +395,13 @@ func (r *Reconciler) patchStatusWithRetryOnConflict(ctx context.Context, app *v1
 	})
 }
 
-func (r *Reconciler) doWorkflowFinish(app *v1beta1.Application, wf workflow.Workflow) error {
+func (r *Reconciler) doWorkflowFinish(ctx monitorContext.Context, app *v1beta1.Application, wf workflow.Workflow) error {
 	if err := wf.Trace(); err != nil {
 		return errors.WithMessage(err, "record workflow state")
 	}
+	if err := wf.Cleanup(ctx); err != nil {
+		return errors.WithMessage(err, "cleanup workflow")
```

Comment:
what will happen if the error occurs?

Reply:
```go
return r.endWithNegativeConditionWithRetry(logCtx, app, condition.ErrorCondition(common.WorkflowCondition.String(), errors.WithMessage(err, "DoWorkflowFinish")), common.ApplicationRunningWorkflow)
```

Reply:
I mean the error should not block the main process